docs/docs/use-cases/flux.md

```diff
@@ -0,0 +1,188 @@
+to set up all necessary Flux structures in the repository.
+Use the following command to do this:
+
+```bash
+GITHUB_USER=<user-name> GITHUB_TOKEN=<token> \
+flux bootstrap github \
+  --owner=$GITHUB_USER \
+  --repository=podtato-head \
+  --branch=main \
+  --path=./clusters/dev \
+  --personal
+```
+
+The bootstrap command above does the following:
+
+* Creates a git repository `podtato-head` on your `GitHub` account.
+* Adds Flux component manifests to the repository -
+creates `./clusters/dev/flux-system/*` structure.
+* Deploys Flux components to your Kubernetes Cluster.
+* Configures Flux components to track the path `./clusters/dev/` in the repository.
+
+## Creating application
+
+Now it's time to add the application together with pre- and
+post-deployments checks to the repository.
+
+Firstly, clone the `podtato-head` repository to your local machine.
+
+Add the following manifests that represent the application into the  `podtato-head.yaml` file
+and store it in the `./apps/dev/` directory of your repository:
+
+```yaml
+{% include "./assets/flux/app.yaml" %}
+```
+
+Additionally, create a `kustomize.yaml` file right next to it:
+
+```yaml
+{% include "./assets/flux/kustomize.yaml" %}
+```
+
+You can commit and push these manifests to the Git repository.
+
+> **Note**
+The application has Keptn pre- and post-deployment tasks defined
+in the manifests.
+This enhances the
+[Flux pre and post-deployment jobs](https://fluxcd.io/flux/use-cases/running-jobs/)
+setup with added observability out of the box.
+
+## Set up continuous delivery for the application
+
+Firstly, we need to create a Flux
+[GitRepository](https://fluxcd.io/flux/components/source/gitrepositories/)
+resource
+pointing to a repository where the application manifests are present.
+In this case, it will be our `podtato-head` repository.
+To create this resource, we use Flux CLI:
+
+```bash
+flux create source git podtato-head \
+  --url=<git-repo-url> \
+  --branch=main \
+  --interval=1m \
+  --export > ./clusters/dev/podtato-head-source.yaml
+```
+
+which results in output similar to:
+
+```yaml
+{% include "./assets/flux/gitrepository.yaml" %}
+```
+
+In the last step, create a Flux
+[Kustomization](https://fluxcd.io/flux/components/kustomize/kustomizations/)
+resource to deploy the `podtato-head` application to the cluster.
+You can create it using the Flux CLI:
+
+```bash
+flux create kustomization podtato-head \
+  --target-namespace=podtato-kubectl \
+  --source=podtato-head \
+  --path="./apps/dev" \
+  --prune=true \
+  --wait=true \
+  --interval=30m \
+  --retry-interval=2m \
+  --health-check-timeout=3m \
+  --export > ./clusters/dev/podtato-head-kustomization.yaml
+```
+
+which results in output similar to:
+
+```yaml
+{% include "./assets/flux/flux-kustomization.yaml" %}
+```
+
+Now commit and push the resources you created in the recent steps.
+After pushing them, Flux picks up the configuration and
+deploys your application into the cluster.
```

Comment:
the above commands could be used directly in an imperative way to let flux push to the repo as opposed to the user running the commands and then having to push manually to the repo.

i guess the commands here are similar to `kubectl apply --dry-run -oyaml > file`

Reply:
Yes I knew about this, but I would prefer (for a beginner user who does not know Flux CLI) to see what exactly is created by the command and then push the resources. I am in favor in more verbose approach than to skip steps (even though automatically)

But if you think it's better that way, I guess we can change it

Reply:
i agree with you. the only alternative would have been to just have the bootstrap command and then do the rest if full gitops fashion by just telling people to create things in the repo. but i think this approach is fine.

Reply:
Since this is a guide, what about adding a Note block that says "The previous commands can instead be used directly in an imperative way... [exactly what Mo wrote above] and then show the commands?